java/pom.xml

```diff
@@ -73,6 +74,7 @@
                 <artifactId>maven-compiler-plugin</artifactId>
                 <version>3.10.1</version>
                 <configuration>
+                    <release>${java.release}</release>
```

Comment:
This ensures that not only are source/target bytecode compatible with Java 8, but it also builds with the Java 8 bootclasspath to ensure no APIs introduced in Java 9+ are accidentally pulled in. It is supported by Java compilers 9+ (we are building with 17 as part of CI).

Ref: https://maven.apache.org/plugins/maven-compiler-plugin/compile-mojo.html#release